### PR TITLE
feat: support replace FieldParser

### PR DESCRIPTION
**Pull request by forsaken628**
**Describe the PR**
Support replace FieldParser. Then we can use this as a libary and write custom feature.

**Relation issue**
no

**Additional context**
no

### CODE REVIEW

**Comment by ubogdan** [comment]
@forsaken628 Looks good until now, congrats. We will need to have positive coverage on parser.go so we can merge it.

**Comment by forsaken628** [comment]
I move a lot code to new file. So it's seem no possible have positive coverage on parser.go.

**Review by ubogdan — APPROVED** [review]
LGTM

**Comment by HYY-yu** [comment]
@forsaken628  The code in parser.go conflicts with my modifications, #1045  😄
But I think you did a great job refactoring the code, and I'll fix mine.

File: parser.go [hunk 1]
```diff
@@ -1053,13 +1057,25 @@ func (parser *Parser) parseStructField(file *ast.File, field *ast.Field) (map[st
 		return map[string]spec.Schema{typeName: *schema}, nil, nil
 	}
 
-	fieldName, schema, err := parser.getFieldName(field)
+	ps := newTagBaseFieldParser(parser, field)
```
Comment (HYY-yu) [inline parser.go:1060]: @forsaken628  It should be:
```
ps := parser.fieldParserFactory(parser, field)
```

**Pull request merged**